angular/src/directives/navigation/router-link-delegate.ts

```diff
@@ -41,7 +41,7 @@ export class RouterLinkDelegateDirective implements OnInit, OnChanges, OnDestroy
   }
 
   private updateTargetUrlAndHref() {
-    if (this.routerLink) {
+    if (this.routerLink?.urlTree) {
```

Comment:
Am I correct in assuming that when `[routerLink]="null"` there is no urlTree created by Angular? Also any backwards incompatible changes to be aware of for Angular 12?

Reply:
Correct, in Angular <= 12, `[routerLink]="null"` was evaluated as `routerLink=""`, but in Angular 13, it is explicitly null. This change was introduced into Angular to allow disabling the router link behavior in app.

Since we are using dependency injection for accessing the router link instance, it should always be defined as long as the `RouterModule` is imported (previous check), but with this change it will make sure the router link is enabled before applying our logic. 

I did verify with the test-app on v12, that the behavior remained the same.

Reply:
Sounds great!